src/Oscoin/Storage/Block/Abstract.hs

```diff
@@ -55,6 +55,10 @@ data BlockStoreReader c tx s m = BlockStoreReader
     -- ^ Get the genesis block.
     , lookupBlock     :: BlockHash c -> m (Maybe (Block c tx (Sealed c s)))
     -- ^ Lookups a 'Block' from the store when given its hash.
+    , lookupBlockByHeight :: Height -> m (Maybe (Block c tx (Sealed c s)))
+    -- ^ Lookup a block, given its 'Height'.
+    , lookupBlocksByHeight :: (Height,Height) -> m (OldestFirst [] (Block c tx (Sealed c s)))
+    -- ^ Returns the blocks (if any) between the height range.
```

Comment:
I think it would be useful to document whether the range is inclusive or exclusive and that we might get fewer blocks than we expect (if the upper bound is too big).

Reply:
d146be09